app/src/main/java/com/rkm/tasky/feature/agenda/screen/AgendaScreen.kt

```diff
@@ -1,20 +1,182 @@
+
+    AgendaScreen(
+        modifier = modifier,
+        showDialog = showDialog,
+        selectedDate = selectedDate,
+        selectedMonth = selectedMonth,
+        sixDay = sixDays,
+        onDateSelected = viewModel::updateSelectedDate,
+        updateSixDays = viewModel::updateSixDays,
+        onShowDatePicker = viewModel::updateShowDialog
+    )
+}
 
 @Composable
-fun AgendaScreenRoot(modifier: Modifier) {
-    Column(
+private fun AgendaScreen(
+    modifier: Modifier,
+    showDialog: Boolean,
+    selectedDate: Long,
+    selectedMonth: String,
+    sixDay: List<DayUiModel>,
+    onDateSelected: (Long) -> Unit,
+    updateSixDays: (Long) -> Unit,
+    onShowDatePicker: () -> Unit
+) {
+    Box(
         modifier = modifier.fillMaxSize(),
-        horizontalAlignment = Alignment.CenterHorizontally,
-        verticalArrangement = Arrangement.Center
+        contentAlignment = Alignment.Center
     ) {
-        Text("Agenda Screen")
+        Column(
+            modifier = modifier.fillMaxSize()
+        ) {
+            TopAppBar(
+                title = { TopBarDateSelector(month = selectedMonth, onClick = onShowDatePicker) },
+                modifier = modifier.statusBarsPadding(),
+                colors = TopAppBarDefaults.topAppBarColors(
+                    titleContentColor = AgendaTopBarTitleColor,
+                    containerColor = AgendaTopBarBackgroundColor
+                )
+            )
+            LazyColumn(
+                modifier = modifier.fillMaxSize()
+                    .background(AgendaMainBodyBackgroundColor)
+                    .clip(RoundedCornerShape(topStart = 30.dp, topEnd = 30.dp))
+                    .background(AgendaMainBodyForegroundColor)
+            ) {
+                item {
+                    SixDaySelector(modifier.padding(top = 16.dp), sixDay, {})
+                }
+            }
+        }
+
+        if (showDialog) {
+            DatePickerModal(
+                selectedDate = selectedDate,
+                onDateSelected = onDateSelected,
+                updateSixDays = updateSixDays,
+                onDismiss = onShowDatePicker
+            )
+        }
     }
+}
+
+@Composable
+private fun TopBarDateSelector(
+    month: String,
+    onClick: () -> Unit,
+) {
+
+    Text(text = month, modifier = Modifier
+        .wrapContentSize()
+        .clickable {
+            onClick()
+        })
+
+}
+
+@Composable
+private fun DatePickerModal(
+    selectedDate: Long,
+    onDateSelected: (Long) -> Unit,
+    updateSixDays: (Long) -> Unit,
+    onDismiss: () -> Unit
+) {
+    val datePickerState = rememberDatePickerState(initialSelectedDateMillis = selectedDate)
+    DatePickerDialog(
+        onDismissRequest = onDismiss,
+        confirmButton = {
+            TextButton(
+                onClick = {
+                    datePickerState.selectedDateMillis?.let {
+                        onDateSelected(it)
+                        updateSixDays(it)
+                    }
```

Comment:
You mentioned that you had an issue with calculating the next 6 days.

How about, in your ViewModel, you react to the start date changing? When it does, automatically calculate the next six days and update your state with the new values. When your state updates, that will trigger your UI to update. This is the power of reactive programming 😄 

What you’ll need is the following:
1. An updated model to represent a single one of your 6 days (I already shared a model in this PR)
2. Either an private MutableStateFlow of your start date or use `snapshotFlow` on your main state model and react to changes to the start date encapsulated in that model
3. Logic to update the 6 days state using the newly changed start date as day 1